src/components/v2/Modal/Modal.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { ReactElement } from 'react';
+import Backdrop from '@mui/material/Backdrop';
+import { Button, Modal as MUIModal, ModalProps } from '@mui/material';
+import Fade from '@mui/material/Fade';
+import Box from '@mui/material/Box';
+import { useModalStyles } from './ModalStyles';
+
+export interface IModalProps extends ModalProps {
+  children: any;
+  className?: string;
+  isOpened: boolean;
+  handleClose: () => void;
+  titleComponent?: ReactElement;
+}
+
+export const Modal: React.FC<IModalProps> = ({
+  children,
+  handleClose,
+  isOpened,
+  titleComponent,
+}) => {
+  const s = useModalStyles({ hasTitleComponent: Boolean(titleComponent) });
+  const closeModal = () => {
+    handleClose();
+  };
+  return (
+    <MUIModal
+      open={isOpened}
+      onClose={closeModal}
+      onBackdropClick={closeModal}
+      closeAfterTransition
+      BackdropComponent={Backdrop}
+      BackdropProps={{
+        timeout: 500,
+      }}
+    >
+      <Fade in={isOpened}>
+        <Box sx={s.box}>
+          <Box sx={s.titleWrapper}>
+            <p style={s.titleComponent}>{titleComponent}</p>
```

Comment:
Let's render the title directly without any wrapper it in a paragraph, since sometimes we'll want to display more than text, like for the supply modal:
<img width="563" alt="Screenshot 2022-03-25 at 10 14 40" src="https://user-images.githubusercontent.com/44675210/160101608-fd6bb428-61dc-4092-953a-d9ac6aeb4174.png">

Reply:
changed paragraph to Box component. cause wrapper is necessary for styling